boredgamers/routes.py

```diff
@@ -57,24 +59,31 @@ def logout():
     return redirect(url_for("home"))
 
 
+
 def save_picture(form_picture):
-    random_hex = secrets.token_hex(8)
-    _, f_ext = os.path.splitext(form_picture.filename)
-    picture_fn = random_hex + f_ext
-    picture_path = os.path.join(app.root_path, "static/profile_pics", picture_fn)
-    output_size = (125, 125)
+    output_size = (200, 200)
     i = Image.open(form_picture)
     i.thumbnail(output_size)
-    i.save(picture_path)
-    return picture_fn
+    buf = io.BytesIO()
+    i.save(buf, format=i.format)
+    bts = buf.getvalue()
+    return bts
+    
+
+@app.route("/picture/<int:userid>")
+def picture(userid):
+    user = User.query.filter_by(id=userid).first()
+    picture = user.image_file
+    return send_file(io.BytesIO(picture), mimetype="image/jpeg")
```

Comment:
I think above we allow users to upload jpgs or pngs. What will happen here if users upload a png file but we set the mime type always to jpg here? Should we either always convert to jpg (`PIL.Image.save` can do this for us) or set the mimetype based on loading the bytes back into a `PIL.Image` here and asking for its `.format`?

Reply:
Right, I missed that!
> set the mimetype based on loading the bytes back into a `PIL.Image` here and asking for its `.format`?
I'm not sure how to do that?

Reply:
You should be able to load the bytes into an `PIL.Image` with

    from PIL import Image
    image = Image.open(bytes)

and then you can ask for `image.format`.